packages/context/test/unit/resolver.test.ts

```diff
@@ -0,0 +1,173 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {Context} from '../../src/context';
+import {inject} from '../../src/inject';
+import {instantiateClass} from '../../src/resolver';
+
+describe('constructor injection', () => {
+  let ctx: Context;
+
+  before(function() {
+    ctx = new Context();
+    ctx.bind('foo').to('FOO');
+  });
+
+  it('can resolve constructor arguments', () => {
+    class TestClass {
+      constructor(@inject('foo') public foo: string) {
+      }
+    }
+
+    const t = instantiateClass(TestClass, ctx) as TestClass;
+    expect(t.foo).to.eql('FOO');
+  });
+});
+
+describe('async constructor injection', () => {
+  let ctx: Context;
+
+  before(function() {
+    ctx = new Context();
+    ctx.bind('foo').to('FOO');
+  });
+
+  it('can resolve constructor arguments', async () => {
+    class TestClass {
+      constructor(@inject('foo') public foo: string) {
+      }
+    }
+
+    const t = await instantiateClass(TestClass, ctx) as TestClass;
```

Comment:
On L123 below, you have the following code:

```ts
const t = await instantiateClass(TestClass, ctx);
```

Can we use the same approach here and avoid the extra `as TestClass` cast?

Please review other tests calling `await instantiateClass` too and make the code consistent.